app/models/host/managed.rb

```diff
@@ -734,8 +734,8 @@ def refresh_global_status!
     save!(:validate => false)
   end
 
-  def refresh_statuses
-    HostStatus.status_registry.each do |status_class|
+  def refresh_statuses(which = HostStatus.status_registry)
+    which.each do |status_class|
```

Comment:
I think the idea came from [build substatus](https://github.com/theforeman/foreman/blob/develop/app/models/host_status/build_status.rb#L26) where we wanted to flip the state with any other state update. It seems to be displayed correctly but it wouldn't be reflected in DB if it was not updated on this refresh_statuses call, so we wouldn't be able to search for host which token expired.

now when we have active job, we could have a job to verify this state in Time.now + $provisioning_interval